pkg/server/render.go

```diff
@@ -39,7 +42,19 @@ func (ctrl *Controller) renderHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	out, err := ctrl.storage.Get(p.gi)
+	var err error
+	var out, leftOut, rghtOut *storage.GetOutput
+
+	leftStartTime, leftEndTime, leftOK := parseRenderRangeParams(r.URL.Query(), "leftFrom", "leftUntil")
+	rghtStartTime, rghtEndTime, rghtOK := parseRenderRangeParams(r.URL.Query(), "rightFrom", "rightUntil")
+
+	isFormatDouble := rghtOK || leftOK
+	if isFormatDouble {
```

Comment:
I generally follow the convention from the original code, where `/render` is `GET` with `from=...&until=...`.

Because `out` is required, hence `from` and `until` are required. So extending the query with `leftFrom`, `leftUntil`, `rightFrom`, `rightUntil` is a reasonable choice.

(personally, I prefer using `POST` in API, but I followed the project convention here)

Reply:
I think we define conventions ourselves :)

I believe that every endpoint/handler should solve exactly one problem, therefore I'd introduce a new one for rending diff. And POST with JSON-encoded parameters would be the most appropriate here, IMO.

Reply:
I'm ok with that. Let's add @petethepig to this conversation so we can have another opinion.

Reply:
@olvrng I generally agree with @kolesnikovae — I think it would make the API simpler if we split it. But I know with these kinds of things there might be other non-obvious reasons to doing things in one function, so I would like to talk to you about this before we change it.